tools/palletjack2pxelinux

```diff
@@ -40,35 +40,155 @@ rescue Errno::EEXIST
   nil
 end
 
-options = {}
+# Generate pxe menus for each OS configuration
+
+def pxemenu_for_netinstall(netinstall)
+  config=netinstall['host.pxelinux.config']
+  label=config.gsub(/-/, ' ')
+  kernel=netinstall['host.pxelinux.kernel']
+  append=netinstall['host.pxelinux.append']
+  variant=netinstall['host.kickstart.variant']
+
+  File.open("#{$options[:output]}/config/#{config}.menu",
+            File::CREAT | File::TRUNC | File::WRONLY, 0644) do |menufile|
+    menufile << "################################################################
+# Automatically generated by PalletJack from
+# #{$options['warehouse']}
+################################################################
+
+UI menu.c32
+PROMPT 0
+MENU INCLUDE /config/graphics.conf
+MENU AUTOBOOT Installing #{label} in # seconds
+TIMEOUT 100
+
+MENU TITLE Install #{label}
+
+LABEL #{config}
+  MENU LABEL ^Install #{label}
+  MENU DEFAULT
+  KERNEL #{kernel}
+  APPEND #{append}
+
+LABEL MainMenu
+  MENU LABEL ^Main Menu
+  KERNEL menu.c32
+  APPEND /config/default
+"
+  end
+end
+
+# Generate pxe menus for each OS version
+
+def pxemenu_for_operating_system(os)
+  config=os['host.pxelinux.config']
+  label=config.gsub(/-/, ' ')
+  kernel=os['host.pxelinux.kernel']
+  append=os['host.pxelinux.append']
+
+  File.open("#{$options[:output]}/config/#{config}.menu",
+            File::CREAT | File::TRUNC | File::WRONLY, 0644) do |menufile|
+    menufile << "################################################################
+# Automatically generated by PalletJack from
+# #{$options['warehouse']}
+################################################################
+
+UI menu.c32
+PROMPT 0
+MENU INCLUDE /config/graphics.conf
+
+MENU TITLE #{label} Installation Menu
+"
+
+    $jack['netinstall',
+          with_all:{'system.os' => config}].each do |netinstall|
+      config=netinstall['host.pxelinux.config']
+      label=config.gsub(/-/, ' ')
```

Comment:
I'm not sure this is a good idea. It's nice and DRY, but it makes for horrible names presented to the human operator. Maybe have the `netinstall` object contain a key `host.kickstart.label` with a nicer string, e.g. `CentOS 7.2.1511 x86_64 (single disk physical machine)` instead of the currently auto-generated `CentOS 7.2.1511 x86_64 Kickstart_sda`.